### ISSUE TITLE
__resolveType is not used when working with graphQL interface type

### ISSUE THREAD

**Issue by Pitchlab** [neo4j/graphql#207]
**Describe the bug**
When working with an interface type named `Content` with a few types , e.g. `Content_A` and `Content_B` (see below), using these in a query on type `Foo` (see below) gives the following error: 

```
"Abstract type \"Content\" must resolve to an Object type at runtime for field \"ContentMatch.content\". Either the \"Content\" type should provide a \"resolveType\" function or each possible type should provide an \"isTypeOf\" function."
```

Trying to remedy this by adding `__resolveType` to a custom resolver does not resolve the issue. Digging into past issues tells me that on the old `neo4j-graphql-js` code, the `makeAugmentedSchema` does **not** call the `__resolveType` code from the custom resolver. I highly suspect that this is the caase also in the new `neo4j-graphql` code. 

**Type definitions**

```
interface Content {
    uri: ID!
}

type Content_A implements Content {
   uri: ID!
   a: String
}

type Content_B implements Content {
   uri: ID!
   b: String
}

type Foo {
   content: [Content] 
}

query FooQuery {
  Foo {
    content {
      uri
      ...on Content_A {
        a
      }
     ...on Content_B {
        b
      }
    }
  }
}
```
**To Reproduce**

See above. 

**Expected behavior**
The `__resolveType` code is called and the error is gone, and we can happily use interfaces. 

**System**
-   OS: MacOS 10.15.1
-   Version: ^1.0.0-beta.2
-   Node.js version: 14.15.1

**Comment by iyosayi** [comment]
Currently experiencing this issue. This has brought the entire project to a halt, is there any workaround for this? Because we rely on using Unions/Interfaces a lot.

**Comment by darrellwarde** [comment]
Hi both, sorry for the radio silence on this one, and sorry to hear you've had to revert back to the old library until this is looked at. 🙁 

We already had our hands full with key bug fixes and features when this was raised, but this certainly looks like a candidate for one of the next major bugs we address. I'll keep you posted!

**Comment by DomVinyard** [comment]
Yeah, running into this same error too, with Unions. We are not seeing this with autogenerated types, only with custom resolver.

Schema:
```
union U = A | B

type A {
  id: ID
}
type B {
  id: ID
}
type User {
  test: U
}
type Query {
   Me: User # custom OGM resolver, returns user 1
}

```

Two queries:
:white_check_mark: Succeeds
```
query {
  Me {
    test {
      ...on A {
        id
      }
    }
  }
}
```

:x: Fails
```
query {
  users(where: { id: 1 }) {
    test {
      ...on A {
        id
      }
    }
  }
}
```
Both queries generate identical Cypher queries:


But the failing state throws:
```
"Abstract type \"U\" must resolve to an Object type at runtime for field \"test\". Either the \"U\" type should provide a \"U\" function or each possible type should provide an \"isTypeOf\" function.",
```

**Comment by darrellwarde** [comment]
Hey everyone, we're going to try to look at this ASAP - looking at finding some time on Friday to dig into this. We'll do a patch release when we come up with a solution.

**Comment by MarianAlexandruAlecu** [comment]
@darrellwarde any update on this?

**Comment by darrellwarde** [comment]
This _should_ now be fixed by #297! 🤞  Our apologies for the delay in finding this one, it was a difficult bug to track down but then very obvious once we've found it! We're going to get it merged and released ASAP.
